### PR TITLE
Updating localization validator to v2.0.0

### PR DESCRIPTION
**Pull request by mishra14**
## Fix
Details: Updating the loc validation tool. The tool has been [updated](https://github.com/NuGet/Entropy/pull/9) in  2 aspects - 

1. Add additional validation step to assert that NuGet dlls in the vsix have all the necessary localized resource dlls in known paths. This will prevent future cases of issue fixed by https://github.com/NuGet/NuGet.Client/commit/1d3aa71b4a437456feb87fb7282c32edb9d882f5
2. Write output files in json format to make them machine parseable

## Testing/Validation
Validation done:  Running validation tool in a separate private branch's [build](https://devdiv.visualstudio.com/DevDiv/NuGet/_build/index?buildId=1628975)

### CODE REVIEW

File: scripts/cibuild/BuildValidator.ps1 [hunk 1]
```diff
@@ -51,6 +51,6 @@ if ($BuildRTM -eq 'false')
         & $NuGetValidator localization --artifacts-path $ArtifactsLocation --output-path $ArtifactsLogOutputDir --comments-path $LocalizationRepository
     }
 
-    # We want to exit the process with success even if there are errors.
-    exit 0
+    # return the exit code from the validator
```
Comment (nkolev92) [inline scripts/cibuild/BuildValidator.ps1:54]: when does this run again?
Would it prevent the tests from running? And if yes, which ones?

**Review by nkolev92 — APPROVED** [review]

File: scripts/cibuild/BuildValidator.ps1 (hunk 1 above)
Comment (mishra14) [reply scripts/cibuild/BuildValidator.ps1:54]: this runs in the unit build step for non rtm build. This should only block tests that use build artifacts.

Comment (nkolev92) [reply scripts/cibuild/BuildValidator.ps1:54]: ok. 
So it's basically a change that could introduce more failures, but probably for the right reason. 

We need to watch out that it doesn't get too noisy.

**Review by rohit21agrawal — CHANGES_REQUESTED** [review]
@mishra14 given that you are just updating the change in the version of the validator, the actual validation changes are a black box to us. I would suggest that you add the validator code to the nuget repository, create a PR for your changes, and once those are approved , only then merge this change.

**Comment by mishra14** [comment]
@rohit21agrawal 🔔

**Review by rohit21agrawal — APPROVED** [review]

**Pull request merged**